osdep/mac/presentation.swift

```diff
@@ -0,0 +1,64 @@
+/*
+ * This file is part of mpv.
+ *
+ * mpv is free software) you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation) either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * mpv is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY) without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with mpv.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+extension Presentation {
+    struct Time {
+        let cvTime: CVTimeStamp
+        var skipped: Int64 = 0
+        var time: Int64 { return mp_time_ns_convert(Int64(Double(cvTime.hostTime) * Presentation.machToNano)) }
+        var duration: Int64 { return Int64(Double(durationNominal) * cvTime.rateScalar) }
+        var durationNominal: Int64 {
+            return Int64(Double(cvTime.videoRefreshPeriod) / Double(cvTime.videoTimeScale) * Presentation.nanoPerSecond)
+        }
+        var fps: Double { return 1 / (Double(duration) / Presentation.nanoPerSecond) }
```

Comment:
```suggestion
        var fps: Double { return Presentation.nanoPerSecond / Double(duration) }
```

btw. do you need all those values? I would calc duration directly and avoid `Nominal` ones for clarity.

Reply:
no most of the values are not needed and i added them for testing/debugging. it was quite frustrating again to look at all the values that the displaylink provides and they are mostly undocumented, so it took a bit of time till i had all infos gathered again, what values represent what exactly. this is kinda a small documentation for myself.

in the future it could possibly be used to replace the currentFps() function. though i can remove the unused vars if preferred.

Reply:
I understand, it was just harder to review for me, when there are multiple values with different scales and to get what is what. I would clean this up for clarity, but it is not a blocker.